File: hibernate-lab/src/main/java/org/example/model/Product.java
package org.example.model;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    private String productName;
    private int unitsInStock;
    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @OneToMany(mappedBy = "product")
    private List<InvoiceItem> invoiceItems = new ArrayList<>();


    public Product(String productName, int unitsInStock) {
        this.productName = productName;
        this.unitsInStock = unitsInStock;
    }

    public Product() {
    }
}

File: lab-3/Product.cs
using System.Collections.ObjectModel;

public class Product
{
    public Product(){   
        this.Invoices = new HashSet<Invoice>();
    }
    public int ProductID { get; set; }
    public String? ProductName { get; set; }
    public int UnitsInStock { get; set; }

    public int? SupplierID { get; set; }
    public Supplier? Supplier { get; set; }

    public virtual ICollection<Invoice>? Invoices { get; set; }
}